### DISCUSSION TITLE
Add ability to pin Airflow Variable(s)/Connection(s) in the UI

### DISCUSSION THREAD

**Discussion by SangwanP** [apache/airflow#36754] [Ideas]
### Description

I'd like to see functionality in the UI to allow pinning some Airflow Variables/Connections at the top of the list.

### Use case/motivation

In our deployment we have a couple of Airflow Variables, and Connections that are central to our operations. We typically have to use filter search in order to reach these Connection/Variable, since we maintain a long list.
It would be nice to have the ability to pin these few important artifacts to the top of the Variable/Connection list in the UI, such that they are readily accessible.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by pateash** [comment]
This looks interesting, let me look into this.

**Comment by bbovenzi** [comment]
assigning you, @pateash

**Comment by eladkal** [comment]
> pin these few important artifacts 

(Just an idea)
Maybe a user can mark his dags/variables/connections as favorites and we can show a separated view of the favorite ones? Then this is also a setting per user rather than some other global setting. In bigger organizations what you call few important artifacts may be a bit tricky as what is important to team A or individual A1 isn't the case for B/B1.

**Comment by msumit** [comment]
IMO having a Recent of DAGs/Connections/Variables etc is another good way to solve this. In general, I find it common that we don't un-favorite things & they just keep on adding up, like the number of bookmarks in our browsers.

**Comment by bbovenzi** [comment]
Maybe just added a updatedAt column and default the sort to the most recently changed?

**Comment by eladkal** [comment]
Most recently changed doesn't necessarily mean most important one but sorting capabilities in general is a good thing to have.

**Comment by potiuk** [comment]
Yeah. I actually hate when the "most recent" list keeps on changing. It does not help you to build "muscle memory"  and "knowing where to search for". Adding "favourite" or "star" is a better idea IMHO. 

And it does not have to be in the database - it could be easily kept in cookie of the user. This will automatically give you "per-person" setup that you would not have to really manage. I see that as a "personalization" layer. Later we could sync the settings for user via the DB, but I think a cookie stored list of favourites should be "good enough" - especially that it would be easy to redo if you change browser/PC.
